src/2048-expert/index.js

```diff
@@ -241,21 +245,46 @@ function keyUpDown() {
     generateNewTile()
 }
 
-//function start new game
+// Check game status
+// check Win condition when 2048 is generated
+function checkWin() {
+    for (let i = 0; i < totalCell; i++) {
+        if (cells[i].innerHTML == 2048) {
+            alert("\t\t You win! \n Your score is " + currentScore)
+            newGame()
+        }
+    }
+} 
+
+// Check Lost
+function checkLost() {
+    let numEmptyCells = 0 
+    for (let i = 0; i < totalCell; i++) {
+        if (cells[i].innerHTML == 0) {
+            numEmptyCells++
+        }
+    }
+    if (numEmptyCells == 0) {
+        alert("\t\t You Lost\n Your score is " + currentScore)
+        newGame()
+    }
+} 
+
+// function start new game
 function newGame() {
     clearBoard()
     generateNewTile()
     generateNewTile()
 }
 
-//function clear Board
+// clear Board
 function clearBoard() {
     cells.forEach(cell => cell.innerHTML = "")
     currentScore = 0
-    ccurrentScoreDisplay .innerHTML = 0
+    currentScoreDisplay.innerHTML = 0
 }
 
-//intialize board
+// intialize board
 createBoard()
 
```

Comment:
I think this part (below) in the `createBoard` function should be moved here instead since it isn't technically part of the board creation. 

```
// intial 2 tiles at random place
generateNewTile()
generateNewTile()
bestScoreDisplay.innerHTML = bestScore
```